### PR TITLE
request + start object invoice

### PR DESCRIPTION
**Pull request by ManuelTrambert**

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//var rp = require('request-promise');
+import requestPromise from 'request-promise';
+
+class Request {
+    constructor(method, url) {
```
Comment (iGitScor) [inline request.js]: You constructor is useless if you decide to create a full static class
Furthermore, if your whole class is static, class is useless

Why don't you instantiate a request and perform it ?

File: request.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        this.method = method;
+        this.url = url;
+
+    }
+
+    static async performRequest(endpoint, method, req, data, api) {
```
Comment (iGitScor) [inline request.js]: Due to context, you don't need to precise what you perform (execute is more appropriate)

File: request.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+//var rp = require('request-promise');
```
Comment (iGitScor) [inline request.js]: Useless comment here
```suggestion
```

File: request.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        if (req && req.query) {
+            req.query = req.query || {};
+            query = Object.assign(query, req.query);
+        }
+        let uri = endpoint;
+        if (!endpoint.startsWith('http')) {
+            uri = api.baseURL + endpoint;
+        }
+        let options = {
+            uri: uri,
```
Comment (iGitScor) [inline request.js]:
```suggestion
            uri,
```

File: request.js (hunk 1 above)
Comment (ManuelTrambert) [reply request.js]: You're right, what is the best solution ? A constructor that instantiate the class + perform it after ? 
or just a method like performRequest?

**Pull request merged**